exercises.ipynb

```diff
@@ -1,142 +1,1013 @@
-    "slideshow": {
-     "slide_type": "-"
+  "nbformat": 4,
+  "nbformat_minor": 0,
+  "metadata": {
+    "colab": {
+      "name": "exercises.ipynb",
+      "version": "0.3.2",
+      "provenance": []
+    },
+    "kernelspec": {
+      "display_name": "Python 3",
+      "language": "python",
+      "name": "python3"
     }
-   },
-   "outputs": [],
-   "source": [
-    "# Our jupyter/datascience-notebook Docker container comes with \n",
-    "# BeautifulSoup4 and requests, both popular libraries!\n",
-    "\n",
-    "from bs4 import BeautifulSoup\n",
-    "import requests"
-   ]
   },
-  {
-   "cell_type": "code",
-   "execution_count": 257,
-   "metadata": {
-    "autoscroll": false,
-    "collapsed": false,
-    "ein.hycell": false,
-    "ein.tags": "worksheet-0",
-    "slideshow": {
-     "slide_type": "-"
+  "cells": [
+    {
+      "metadata": {
+        "autoscroll": false,
+        "ein.hycell": false,
+        "ein.tags": "worksheet-0",
+        "id": "xI7YTPdcANQl",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Our jupyter/datascience-notebook Docker container comes with \n",
+        "# BeautifulSoup4 and requests, both popular libraries!\n",
+        "\n",
+        "from bs4 import BeautifulSoup\n",
+        "import requests"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "autoscroll": false,
+        "ein.hycell": false,
+        "ein.tags": "worksheet-0",
+        "id": "lV7cffNMANQq",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "START_URL = 'https://brickset.com/sets/year-2016'"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "autoscroll": false,
+        "ein.hycell": false,
+        "ein.tags": "worksheet-0",
+        "id": "Igjod2zMANQv",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def get_titles(soup):    \n",
+        "    \"\"\" Returns a list of titles on the page \"\"\"\n",
+        "    # the \"soup\" parameter is of the type that is\n",
+        "    # returned by Beautiful Soup when it parses HTML.\n",
+        "    # The function should then use the object to\n",
+        "    # extract a list of titles (of the lego sets)\n",
+        "    # Grab each product\n",
+        "    metas = soup.findAll(\"div\", {\"class\":\"meta\"})\n",
+        "    # Grab each title\n",
+        "    titles = []\n",
+        "    for meta in metas:\n",
+        "        titles.append(meta.h1.text)  \n",
+        "    # Lookup the documentation for Beautiful Soup\n",
+        "    # Figure out how to select the text of the title\n",
+        "    # of each legoset. A title should look like: \n",
+        "    # \"10252: Volkswagen Beetle\"\n",
+        "    def catch_index_error():\n",
```

Comment:
This is clunky! The function is only called once, and has no inputs (it just uses the variables outside of it). 

```{py}
try: 
    next_link = soup.select('li.next a')[0]['href']
    return titles + parse_bricks(next_link)
except IndexError:
    return titles
```